src/popup.c

```diff
@@ -58,12 +58,37 @@ void ShutdownPopup(void)
    }
 }
 
+/** Calculate dimensions of a popup window given the popup text. */
+char** MeasurePopupText(const char *text, int *width, int *height, int *rows) {
+    char **result = NULL;
+    char *ptr     = NULL;
+
+    *width  = 0;
+    *height = 1;
+    *rows   = 0;
+
+    for (ptr = strtok(strdup(text), "\n"); ptr != NULL; ptr = strtok(NULL,"\n"))
+    {
+        int current_width = GetStringWidth(FONT_POPUP, ptr) + 9;
+        if(*width < current_width)
+            *width = current_width;
+        *height = *height + GetStringHeight(FONT_POPUP) + 1;
+        result = (char**)Reallocate(result, sizeof(ptr)+sizeof(result));
+        result[(*rows)++] = strdup(ptr);
+    }
+
```

Comment:
If you compile JWM with `--enable-debug`, you get some problems related to the use of `strdup`.  There is a function in the JWM source, `CopyString` to do this and keep the memory leak checker happy.  The leak checker will also complain about `Reallocate` being used with a `NULL` pointer, which really isn't  serious, so it might even be worth updating the `Reallocate` implementation to accept that.